ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/PureConfigReaderWriter.scala

```diff
@@ -0,0 +1,319 @@
+          CannotConvert(
+            unknownSeeding,
+            Seeding.getClass.getName,
+            s"Seeding is neither ${Seeding.Strong.name}, ${Seeding.Weak.name}, nor ${Seeding.Static.name}: ${unknownSeeding}",
+          )
+        )
+    }
+
+  implicit val contractIdSeedingWriter: ConfigWriter[Seeding] = ConfigWriter.toString(_.name)
+
+  implicit val userManagementConfigReader: ConfigReader[UserManagementConfig] =
+    deriveReader[UserManagementConfig]
+  implicit val userManagementConfigWriter: ConfigWriter[UserManagementConfig] =
+    deriveWriter[UserManagementConfig]
+
+  implicit val authServiceConfigUnsafeJwtHmac256Reader
+      : ConfigReader[AuthServiceConfig.UnsafeJwtHmac256] =
+    deriveReader[AuthServiceConfig.UnsafeJwtHmac256]
+  implicit val authServiceConfigJwtEs256CrtReader: ConfigReader[AuthServiceConfig.JwtEs256Crt] =
+    deriveReader[AuthServiceConfig.JwtEs256Crt]
+  implicit val authServiceConfigJwtEs512CrtReader: ConfigReader[AuthServiceConfig.JwtEs512Crt] =
+    deriveReader[AuthServiceConfig.JwtEs512Crt]
+  implicit val authServiceConfigJwtRs256CrtReader: ConfigReader[AuthServiceConfig.JwtRs256Crt] =
+    deriveReader[AuthServiceConfig.JwtRs256Crt]
+  implicit val authServiceConfigJwtRs256JwksReader: ConfigReader[AuthServiceConfig.JwtRs256Jwks] =
+    deriveReader[AuthServiceConfig.JwtRs256Jwks]
+  implicit val authServiceConfigWildcardReader: ConfigReader[AuthServiceConfig.Wildcard.type] =
+    deriveReader[AuthServiceConfig.Wildcard.type]
+  implicit val authServiceConfigReader: ConfigReader[AuthServiceConfig] =
+    deriveReader[AuthServiceConfig]
+
+  implicit val authServiceConfigJwtEs256CrtWriter: ConfigWriter[AuthServiceConfig.JwtEs256Crt] =
+    deriveWriter[AuthServiceConfig.JwtEs256Crt]
+  implicit val authServiceConfigJwtEs512CrtWriter: ConfigWriter[AuthServiceConfig.JwtEs512Crt] =
+    deriveWriter[AuthServiceConfig.JwtEs512Crt]
+  implicit val authServiceConfigJwtRs256CrtWriter: ConfigWriter[AuthServiceConfig.JwtRs256Crt] =
+    deriveWriter[AuthServiceConfig.JwtRs256Crt]
+  implicit val authServiceConfigJwtRs256JwksWriter: ConfigWriter[AuthServiceConfig.JwtRs256Jwks] =
+    deriveWriter[AuthServiceConfig.JwtRs256Jwks]
+  implicit val authServiceConfigUnsafeJwtHmac256Writer
+      : ConfigWriter[AuthServiceConfig.UnsafeJwtHmac256] =
+    deriveWriter[AuthServiceConfig.UnsafeJwtHmac256]
+  implicit val authServiceConfigWildcardWriter: ConfigWriter[AuthServiceConfig.Wildcard.type] =
+    deriveWriter[AuthServiceConfig.Wildcard.type]
+  implicit val authServiceConfigWriter: ConfigWriter[AuthServiceConfig] =
+    deriveWriter[AuthServiceConfig]
+
+  implicit val partyConfigurationReader: ConfigReader[PartyConfiguration] =
+    deriveReader[PartyConfiguration]
+  implicit val partyConfigurationWriter: ConfigWriter[PartyConfiguration] =
+    deriveWriter[PartyConfiguration]
+
+  implicit val commandConfigurationReader: ConfigReader[CommandConfiguration] =
+    deriveReader[CommandConfiguration]
```

Comment:
```suggestion
  implicit val commandConfigurationReader: ConfigConvert[CommandConfiguration] =
    deriveConvert[CommandConfiguration]
```
We can shave some lines if we derive both at the same time where we need both readers and writer

Reply:
very good point! thank you!